ccs811: publish firmware version; log bootloader and HW version; fix a bug

Habbie opened the pull request:
# What does this implement/fix? 

CCS811 modules have firmware, and many of them run version 1.0 or 1.1. The vendor has published 2.0.0 and 2.0.1 which are recommended over the 1.x versions. This PR adds logging of the HW revision, the bootloader version, and the loaded firmware version.

This PR fixes a typo in an enum.

This PR fixes a bug where the measurement interval on the CCS811 module was configured inappropriately in relation to the configured update interval in ESPHome, causing missed updates.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** I'm not aware of any.

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** https://github.com/esphome/esphome-docs/pull/1336

## Test Environment

- [ ] ESP32
- [x] ESP8266

## Example entry for `config.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example config.yaml
sensor:
  - platform: ccs811
    eco2:
      name: "CCS811 eCO2 Value"
    tvoc:
      name: "CCS811 Total Volatile Organic Compound"
    version:
      name: "CCS811 Firmware Version"
    address: 0x5A
    update_interval: 5s

```

Compared to the example config in the docs, the `version` key is added. Unlike the other two sensors for ccs811, `version` is optional.

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

I have not yet looked into the test framework. I will do that soon.

I did not test this with an actual HomeAssistant install yet. I'm not worried about it, but I'll do it soon. I may then find out that `ICON_RESTART` was a weird choice.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

Habbie committed:
ccs811: log versions and publish firmware version as sensor, plus other changes:

- fix reversed interval logic
before this commit, a 30s interval would cause 60s interval configured
on the sensor, leading to a failed read 50% of the time

- fix typo in enum name
- add CODEOWNERS

Habbie commented:
Tested with HA now:

![image](https://user-images.githubusercontent.com/525838/125208532-59d2f700-e293-11eb-8888-3c2295fa5362.png)

agners commented on esphome/components/ccs811/ccs811.cpp:
```diff
@@ -76,6 +78,36 @@ void CCS811Component::update() {
 
+  auto hardware_version_data = this->read_bytes<1>(0x21);
+  auto bootloader_version_data = this->read_bytes<2>(0x23);
+  auto application_version_data = this->read_bytes<2>(0x24);
```
Those versions do not change at runtime, so we should only read it on setup and not in every update.

Habbie replied:
Oops, that was supposed to be inside `  if (this->status_has_warning()) {` so it only happens at boot and if somebody disconnects+reconnects the sensor, which I wanted to support.

Shall I fix it (move it inside the `if`), or move it to setup?

agners replied:
I still think setup would make more sense.

Habbie replied:
Thanks, I should have commented after I also discussed this on Discord. I will move it to setup.

Habbie replied:
moved!

Habbie commented:
Rebased to fix a conflict.

Habbie commented:
Docs in https://github.com/esphome/esphome-docs/pull/1336

OttoWinter commented on esphome/components/ccs811/ccs811.h:
```diff
@@ -43,6 +44,7 @@ class CCS811Component : public PollingComponent, public i2c::I2CDevice {
 
   sensor::Sensor *co2_{nullptr};
   sensor::Sensor *tvoc_{nullptr};
+  sensor::Sensor *version_{nullptr};
```
```suggestion
  text_sensor::TextSensor *version_{nullptr};
```

Nit: A version number should really be published as a text sensor - will require a couple of smaller changes too:

* Add `AUTO_LOAD = ["text_sensor"]` to `ccs811/sensor.py`
* Convert `sensor.sensor_schema` to use `text_sensor.TEXT_SENSOR_SCHEMA`
* Use `sens = cg.new_Pvariable(config[CONF_VERSION][CONF_ID])` and `await text_sensor.register_text_sensor(sens, config[CONF_VERSION])` to declare the text sensor.
* Publish state using `publish_state` as before, but instead pass in a formatted `std::string`

Habbie replied:
I was overthinking your list of changes but @ssieb put me back on the right path. It's a text sensor now.

Habbie commented:
Changed it to be a text sensor. Then fixed the icon again. I accidentally replaced async/await with yield in the text sensor, I also fixed that now.

Habbie commented:
Suggestion applied, CI is happy again. If this looks good, I'll update the docs (https://github.com/esphome/esphome-docs/pull/1336) to explain the version string.

Habbie commented:
I updated the docs anyway.

alexiri commented:
Some CCS811 breakout boards have an `nWAKE` pin to control when the sensor is awake:

> nWAKE can either be tied to GND, in which case the CCS811 is always awake (using more power). Alternatively nWAKE can be tied to a GPIO pin of the micro, the CCS library will then use this pin to wake-up the micro when needed.

(https://github.com/maarten-pennings/CCS811)

Would it be possible to implement that here as well? Or is there some way in esphome we can trigger a GPIO before performing a sensor reading?

jesserockz commented on esphome/components/ccs811/sensor.py:
```diff
@@ -13,15 +15,21 @@
 CCS811Component = ccs811_ns.class_(
     "CCS811Component", cg.PollingComponent, i2c.I2CDevice
 )
+# CCS811VersionInfo = ccs811_ns.class_(
+#     "CCS811VersionInfo", text_sensor.TextSensor
+# )
```
No commented code please

jesserockz reviewed (changes requested):
One thing, otherwise looks good

Habbie commented:
Squashed and rebased.
